hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -152,27 +160,35 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
             OMException.ResultCodes.BUCKET_ALREADY_EXISTS);
       }
 
-      LOG.debug("created bucket: {} in volume: {}", bucketName, volumeName);
-      omMetrics.incNumBuckets();
-
       // Update table cache.
       metadataManager.getBucketTable().addCacheEntry(new CacheKey<>(bucketKey),
           new CacheValue<>(Optional.of(omBucketInfo), transactionLogIndex));
 
-      // return response.
+
+    } catch (IOException ex) {
+      exception = ex;
+    } finally {
+      metadataManager.getLock().releaseBucketLock(volumeName, bucketName);
+      metadataManager.getLock().releaseVolumeLock(volumeName);
+
+      // Performing audit logging outside of the lock.
+      auditLog(auditLogger, buildAuditMessage(OMAction.CREATE_BUCKET,
+          omBucketInfo.toAuditMap(), exception, userInfo));
+    }
```

Comment:
Any reason for having the auditLog in the finally block?

Reply:
Yes, it not to do logging/expensive things inside the lock. As after adding the response to cache, we can release the lock, so that other threads waiting for the lock can acquire it. (In future we don't see performance things because of audit log we are seeing some performance issues.) 

More on a side note, doing audit logging outside lock will not cause any side effects in my view. Let me know if anything I am missing here.

Reply:
Yes I meant why not outside the finally block as generally we keep the must be done things (like lock releases) in the finally block.

Reply:
Yes it can be done. Will update it if you prefer that way?

Reply:
Done. Addressed it.